### PR TITLE
Update gas for storage occupation

### PR DESCRIPTION
**Pull request by grarco**
## Describe your changes

Increases the gas cost component for storage occupation and ties this cost the other ones so that it's not free-floating anymore (which could cause a divergence if the other costs are changed).

Removes the redundant `new_from_sub_limit` constructor for `TxGasMeter`.

## Indicate on which release or other PRs this topic is based on

#3428 (diff for review: https://github.com/anoma/namada/pull/3510/files/209da0c6ad9cd456a5a6634ec25cc45408e0b72c..33a6cc9437692ccb1ef24e58f2536ac2dfdeb8f3)

## Checklist before merging to `draft`
- [x] I have added a changelog
- [x] Git history is in acceptable state

### CODE REVIEW

**Comment by grarco** [comment]
Let's wait for the CI to come back online before reviewing this cause there's a change we might have a few tests failing due to gas

File: crates/gas/src/lib.rs
Comment (brentstone) [inline crates/gas/src/lib.rs:64]: why did you write the argument as `(1 + 10`? Do each of these numbers account for some different effect?

**Review by brentstone — APPROVED** [review]
LGTM, modulo response to the `(1 + 10)` comment

File: crates/gas/src/lib.rs
Comment (grarco) [reply crates/gas/src/lib.rs:64]: They do: the 1 represent the actual cost of storage latency, the 10 component instead is just an arbitrary multiplier that I use to compute an actual storage occupation gas cost based on the storage latency. They are not really tied but we cannot leave the storage occupation gas cost free-floating because we'd end up with the same issue this Pr tries to solve. So we need a way to tie that cost to some other gas cost that is actually based on a benchmark and the storage latency is the closest one

Comment (brentstone) [reply crates/gas/src/lib.rs:64]: cool would you mind describing this a bit in a comment / docstring?

Comment (grarco) [reply crates/gas/src/lib.rs:64]: It's described right above

Comment (brentstone) [reply crates/gas/src/lib.rs:64]: Ok I think I was having trouble understanding the comment at first and was seeking clarity, but I think it's fine now. Thanks.

**Pull request merged**
